admin-metrics/src/main/kotlin/com/epam/drill/admin/metrics/service/impl/MetricsServiceImpl.kt

```diff
@@ -470,22 +461,26 @@ class MetricsServiceImpl(
     override suspend fun refreshMaterializedViews() = coroutineScope {
         logger.debug { "Refreshing materialized views..." }
 
+        val buildsViewJob = async { refreshMaterializedView(buildsView) }
         val methodsViewJob = async { refreshMaterializedView(methodsView) }
-        val buildsViewJob = async { waitFor(methodsViewJob).run { refreshMaterializedView(buildsView) } }
-        val methodsCoverageViewJob = async { waitFor(methodsViewJob).run { refreshMaterializedView(methodsCoverageView) } }
-        val buildsComparisonViewJob = async { waitFor(methodsViewJob, buildsViewJob).run { refreshMaterializedView(buildsComparisonView) } }
-        val testedBuildsComparisonViewJob = async { waitFor(methodsViewJob).run { refreshMaterializedView(testedBuildsComparisonView) } }
-        val buildsCoverageViewJob = async { waitFor(methodsCoverageViewJob).run { refreshMaterializedView(buildsCoverageView) } }
-        val testSessionBuildsCoverageViewJob = async { waitFor(methodsViewJob).run { refreshMaterializedView(testSessionBuildsCoverageView) } }
-        waitFor(buildsComparisonViewJob, testedBuildsComparisonViewJob, buildsCoverageViewJob, testSessionBuildsCoverageViewJob)
+        val buildMethodsViewJob = async { refreshMaterializedView(buildMethodsView) }
+        val testLaunchesViewJob = async { refreshMaterializedView(testLaunchesView) }
+        val testDefinitionsViewJob = async { refreshMaterializedView(testDefinitionsView) }
+        val testSessionsViewJob = async { refreshMaterializedView(testSessionsView) }
+        val methodCoverageViewJob = async { waitFor(buildsViewJob, methodsViewJob, buildMethodsViewJob,
+            testLaunchesViewJob, testDefinitionsViewJob, testSessionsViewJob).run { refreshMaterializedView(methodCoverageView) } }
+        val methodSmartCoverageViewJob = async { waitFor(methodCoverageViewJob).run { refreshMaterializedView(methodSmartCoverageView) } }
+        val testSessionBuildsViewJob = async { waitFor(methodCoverageViewJob).run { refreshMaterializedView(testSessionBuildsView) } }
+
+        waitFor(testSessionBuildsViewJob, methodSmartCoverageViewJob)
```

Comment:
why wait for these specific jobs, but not the other ones?

Reply:
By this point the code has already waited for other jobs, so only the remaining ones are waiting here